### PR TITLE
A golang package to get IP address for Mesos containers.

### PR DESCRIPTION
**Pull request by asridharan**
This change is [Reviewable](https://reviewable.io/reviews/dcos/dcos-cni/7)

### CODE REVIEW

**Comment by asridharan** [comment]
@benclarkwood can you take a look at this PR.
The library is sitting in pkg/mesos/ip.go . Rest of the commits are infra required for testing this package.

File: pkg/mesos/ip.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+func ContainerIP() (containerIP net.IP, err error) {
+	// NOTE: Make sure to maintain the order of the environment variables.
+	envs := [2]string{"MESOS_CONTAINER_IP", "LIBPROCESS_IP"}
+	for _, env := range envs {
+		ip := os.Getenv(env)
+		if ip != "" {
+			containerIP = net.ParseIP(ip)
+			if containerIP == nil {
+				err = fmt.Errorf("Invalid %s found: %s", env, ip)
+				return
```
Comment (nickbp) [inline pkg/mesos/ip.go:30]: should this try to proceed to LIBPROCESS_IP if MESOS_CONTAINER_IP is invalid?

File: pkg/mesos/ip.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	// The expectation is that either the `MESOS_CONTAINER_IP` or the `LIBPROCESS_IP` should be set.
+	if containerIP == nil {
+		err = fmt.Errorf("Cannot find container IP address. Either `MESOS_CONTAINER_IP` or `LIBPROCESS_IP` should be set")
+		return
+	}
+
+	if containerIP.To4() == nil {
+		err = fmt.Errorf("Only IPv4 addresses are currently supported:%s", containerIP)
```
Comment (nickbp) [inline pkg/mesos/ip.go]: should this be part of the validity checks inside of the loop?

File: pkg/mesos/ip.go (hunk 1 above)
Comment (asridharan) [reply pkg/mesos/ip.go:30]: No if `MESOS_CONTAINER_IP` is set we expect it to be valid else this a bug in Mesos.

File: pkg/mesos/ip.go (hunk 2 above)
Comment (asridharan) [reply pkg/mesos/ip.go]: Yeah can move it inside the loop.

**Comment by asridharan** [comment]
@nickbp addressed your comments. Can you take another look?

**Review by nickbp — APPROVED** [review]

**Pull request merged**